src/bindings/swig/python3/kdb.i

```diff
@@ -211,6 +215,19 @@
    return new kdb::KeySet(alloc, KS_END);
   }
 
+  kdb::KeySet *__copy__() {
+    return new kdb::KeySet(*$self);
+  }
+
+  kdb::KeySet *__deepcopy__(PyObject *memo) {
+    (void) PyDict_Check(memo);
+    ssize_t size = $self->size();
+    kdb::KeySet *ks = new kdb::KeySet(size, KS_END);
+    for(cursor_t cursor = 0; cursor < size; ++cursor)
+      ks->append($self->at(cursor)->dup());
```

Comment:
why not using ksDup? ksDup will have optimizations in future, e.g. use the same hashmap without rebuilding it